catalog/security/catalog-security-resourceplugin/src/main/java/org/codice/ddf/catalog/security/ResourceUriPolicy.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Override
+    public PolicyResponse processPreDelete(List<Metacard> metacards,
+            Map<String, Serializable> properties) throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostDelete(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreQuery(Query query, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostQuery(Result input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreResource(ResourceRequest resourceRequest)
+            throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostResource(ResourceResponse resourceResponse, Metacard metacard)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    public CatalogFramework getCatalogFramework() {
+        return catalogFramework;
+    }
+
+    public void setCatalogFramework(CatalogFramework catalogFramework) {
+        this.catalogFramework = catalogFramework;
+    }
+
+    public String[] getCreatePermissions() {
+        return createPermissions == null ? null : createPermissions.clone();
+    }
+
+    public void setCreatePermissions(String[] createPermissions) {
+        this.createPermissions = createPermissions == null ? null : createPermissions.clone();
+    }
+
+    public FilterBuilder getFilterBuilder() {
+        return filterBuilder;
+    }
+
+    public void setFilterBuilder(FilterBuilder filterBuilder) {
+        this.filterBuilder = filterBuilder;
+    }
+
+    public String[] getUpdatePermissions() {
+        return updatePermissions == null ? null : updatePermissions.clone();
+    }
+
+    public void setUpdatePermissions(String[] updatePermissions) {
+        this.updatePermissions = updatePermissions == null ? null : updatePermissions.clone();
+    }
+
+    protected Metacard getMetacardFromCatalog(String id) {
+        Filter filter = filterBuilder.attribute(Metacard.ID)
+                .is()
+                .equalTo()
+                .text(id);
+        Query query = new QueryImpl(filter);
+        QueryRequest queryRequest = new QueryRequestImpl(query);
+        QueryResponse queryResponse;
+        try {
+            queryResponse = catalogFramework.query(queryRequest);
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            throw new RuntimeException(e);
+        }
+        Result queryResult = queryResponse.getResults()
+                .get(0);
```

Comment:
Don't blindly do a `.get(0)` Without checking the size. We can return results that have a zero sized result list